index.bs

```diff
@@ -1294,44 +1296,112 @@ A {{ReceiveStream}} has the following internal slots.
 |internalStream| and a {{WebTransport}} |transport|, run these steps:
 
 1. Let |stream| be a [=new=] {{ReceiveStream}}, with:
-    : [=[[InternalStream]]=]
+    : [=ReceiveStream/[[InternalStream]]=]
     :: |internalStream|
-1. [=ReadableStream/Set up=] |stream| with TBD.
+    : [=ReceiveStream/[[Transport]]=]
+    :: |transport|
+1. Let |pullAlgorithm| be an action that [=pulls bytes=] from |stream|.
+1. Let |cancelAlgorithm| be an action that [=ReceiveStream/cancels=] |stream|.
+1. [=ReadableStream/Set up with byte reading support=] |stream| with
+   [=ReadableStream/set up with byte reading support/pullAlgorithm=] set to |pullAlgorithm| and
+   [=ReadableStream/set up with byte reading support/cancelAlgorithm=] set to |cancelAlgorithm|.
 1. [=set/Add=] |stream| to |transport|'s [=[[ReceiveStreams]]=].
 1. Return |stream|.
 
 </div>
 
-When a {{ReceiveStream}} receives a message from the remote side aborting the
-stream (for QUIC, that message is an RESET_STREAM frame), the user agent MUST
-queue a task to run the following:
-      1. Let |stream| be the {{ReceiveStream}} object for which the reset
-         message was received.
-      1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-         from.
-      1. Let |reason| be the value from the message from the remote side.
-      1. [=ReadableStream/Error=] |stream| with |reason|.
-
-The {{ReceiveStream}}'s <dfn>cancelAlgorithm</dfn> is:
-     1. Let |stream| be the {{ReceiveStream}} object.
-     1. Let |transport| be the {{WebTransport}}, which the |stream| was created
-        from.
-     1. Let |reason| be the first argument, if it is numeric, or 0.
-     1. If |stream| has received a RESET_STREAM frame, abort these steps.
-     1. [=In parallel=], [=send STOP_SENDING=] over |stream|'s
-        [=WebTransport stream=] with |reason| as the Application Protocol Error
-         Code.
+<div algorithm>
+
+To <dfn for="ReceiveStream">pull bytes</dfn> from a {{ReceiveStream}} |stream|, run these steps.
+
+1. Let |controller| be |stream|'s [=ReadableStream/[[controller]]=].
+1. Assert: |controller| is a {{ReadableByteStreamController}}.
+1. Let |transport| be |stream|'s [=ReceiveStream/[[Transport]]=].
+1. Let |internalStream| be |stream|'s [=ReceiveStream/[[InternalStream]]=].
+1. Let |promise| be a new promise.
+1. Let |view| be [=ReadableStream/current BYOB request view=] for |stream|.
+1. If |view| is null, set |view| to a [=new=] {{Uint8Array}} with an [=implementation-defined=]
+   size. If allocating the {{Uint8Array}} fails, return a promise [=rejected with=] a
+   {{RangeError}}.
+1. Let |bytes| be the byte sequence |view| represents.
+1. Assert: |bytes|'s length equals to |view|.\[[ByteLength]].
+1. Let |arrayBuffer| be |view|.\[[ViewedArrayBuffer]].
+1. Let |offset| be |view|.\[[ByteOffset]].
+1. Return |promise| and run the remaining steps [=in parallel=].
+1. [=stream/receive|Read bytes=] into |bytes|, from |internalStream|. Wait until either at least one
```

Comment:
I would suggest phrasing this in terms of https://heycam.github.io/webidl/#arraybufferview-write , i.e. something like

```html
[=ArrayBufferView/write=] the bytes that area [=stream/receive|read=] from |internalStream| into |view|.
```

You may also want to require somehow that the implementation not receive more bytes than |view|'s [byte length](https://heycam.github.io/webidl/#buffersource-byte-length) ? It's kind of implied, but not obvious.

Reply:
Done.